src/util.rs

```diff
@@ -270,35 +345,40 @@ fn MAKELANGID(p: u32, s: u32) -> u32 {
 
 /// Returns a a human readable error message from a windows error code
 pub fn format_message(error_code: u32) -> Result<String, Box<dyn std::error::Error>> {
-    let buf = PWSTR::null();
+    let buf: *mut u16 = std::ptr::null_mut();
 
     let chars_written = unsafe {
         FormatMessageW(
             FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_ALLOCATE_BUFFER,
-            None,
+            std::ptr::null_mut(),
             error_code,
             MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
-            PWSTR(&buf as *const windows::core::PWSTR as *mut u16),
+            &buf as *const windows_sys::core::PWSTR as *mut u16,
             0,
-            None,
+            std::ptr::null_mut(),
         )
     };
     if chars_written == 0 {
-        return Err(get_last_error().into());
+        return Ok(get_last_error()?);
     }
-    let result = unsafe { buf.to_string()? };
-    if let Err(v) = unsafe { LocalFree(HLOCAL(buf.as_ptr() as *mut _)) } {
-        log::trace!("LocalFree \"{}\"", v);
+    let result = win_pwstr_to_string(buf)?;
+    if unsafe { !LocalFree(buf as *mut _).is_null() } {
```

Comment:
Still need a comment here like `Win32 returns the same handle if LocalFree fails`. The logic is difficult to follow without leaving your editor and looking up the docs for LocalFree